src/plugins/data/common/es_query/kuery/functions/is.ts

```diff
@@ -86,13 +95,8 @@ export function toElasticsearchQuery(
     });
   }
 
-  const isExistsQuery = valueArg.type === 'wildcard' && (value as any) === '*';
-  const isAllFieldsQuery =
-    (fullFieldNameArg.type === 'wildcard' && ((fieldName as unknown) as string) === '*') ||
-    (fields && indexPattern && fields.length === indexPattern.fields.length);
-  const isMatchAllQuery = isExistsQuery && isAllFieldsQuery;
-
-  if (isMatchAllQuery) {
+  if (isExistsQuery && fields && fields.length === indexPattern?.fields.length) {
```

Comment:
fields?.length

Reply:
before we used OR betwen these conditions
`wildcardStuff || fields.length === indexpatternfields.length`

Reply:
You're right, I will use an OR here. The special case that I missed, but was in the previous logic, is:

1. There is no index pattern
2. The query is a wildcard query
3. It should shortcut and be a match_all